src/securejoin.rs

```diff
@@ -379,16 +397,16 @@ pub(crate) fn handle_securejoin_handshake(
             // it just ensures, we have Bobs key now. If we do _not_ have the key because eg. MitM has removed it,
             // send_message() will fail with the error "End-to-end-encryption unavailable unexpectedly.", so, there is no additional check needed here.
             // verify that the `Secure-Join-Invitenumber:`-header matches invitenumber written to the QR code
-            let invitenumber = match mimeparser.get(HeaderDef::SecureJoinInvitenumber) {
+            let invitenumber = match mime_message.get(HeaderDef::SecureJoinInvitenumber) {
                 Some(n) => n,
                 None => {
-                    warn!(context, "Secure-join denied (invitenumber missing).",);
-                    return Ok(ret);
+                    warn!(context, "Secure-join denied (invitenumber missing)");
+                    return Ok(HandshakeMessage::Ignore);
```

Comment:
Can use [`ok_or_else`](https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or_else) and `?` instead.

Reply:
No, because Err is reserved for application errors.  This is not an error with calling the function, it's an error from the contact who sent this.  For receive_imf() this message was handled correctly, and it should ignore it.

I did first create a version which used Err for protocol errors as well which indeed gives a slightly nicer reading code.  I think I'd end up with wrapping this function into something which distinguishes protocol-errors and turns them into HandshakeMessage::Ignore for receive_imf() while propagating application-errors as Err to receive_imf().  But I ended up concluding that's better for future refactors of this code.